src/AggregateFunctions/AggregateFunctionMap.h

```diff
@@ -228,6 +228,11 @@ class AggregateFunctionMap final
         destroyImpl<false>(place);
     }
 
+    constexpr bool hasTrivialDestructor() const override
+    {
+        return std::is_trivially_destructible_v<Data> && nested_func->hasTrivialDestructor();
```

Comment:
I can't add a bad test case because it wasn't working incorrectly. But I believe this wasn't working incorrectly by pure chance, and I did have to double check so it was really unclear if it was working correctly.

The default implementation (`std::is_trivially_destructible_v<Data>`) was always false because `AggregateFunctionMapCombinatorData` is not trivially destructible. But if that data had been trivially destructible and the nested function wasn't then we'd be leaking memory. For example:

```sql
SELECT countDistinctMap(map(CAST(number, 'UInt256'), CAST(number, 'UInt256'))) from numbers(1)
```

In that map combinator:
* `std::is_trivially_destructible_v<Data>`: False
* `nested_func->hasTrivialDestructor();`: False

If somebody changed `AggregateFunctionMapCombinatorData` and it became trivially destructible then the old check would say that the combinator was trivially destructible without considering the internal function. In the case where the nested function has a destructor that must be called, a leak might have happened.

So this isn't a bug fix per se, more like a prophylactic bug fix ;D